### PR TITLE
Update minSDK and dependencies

### PR DESCRIPTION
**Pull request by domesticmouse**

### CODE REVIEW

**Comment by domesticmouse** [comment]
PTAL @kevmoo

**Comment by kevmoo** [comment]
What does this get us?

**Comment by domesticmouse** [comment]
Impetus for this sweep was https://github.com/dart-lang/dart-tutorials-samples/pull/49#discussion_r295938779

**Comment by domesticmouse** [comment]
Travis breakage is due to https://github.com/dart-lang/sdk/issues/37324

**Comment by domesticmouse** [comment]
@kevmoo I thought more about your question about what this gets us and I have a different answer now that I'm not knee deep in changes.

Modifying the default getting started experience for Stagehand users to include the latest version of the language means that if they are trying out things like UI as Code, they don't get confusing lint warnings and start wondering if everything is broken.

I'm curious what you see as the upside to keeping minSDK at a lower bound? This is going to be a very real problem once we want the community to start getting the upside of NNBD.

**Comment by kevmoo** [comment]
Language features is a good reason. Just checking that we have one. 😄

**Comment by domesticmouse** [comment]
@kevmoo can you poke Travis? It should rebuild clean now that `dart_internal` has been updated.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 3.3.2
+- Updated minimum Dart SDK to 2.3
+- Updated dependency versions across all templates
```
Comment (kwalrath) [inline CHANGELOG.md]: Very Important Comment: We add a period after each list item.

Comment (domesticmouse) [reply CHANGELOG.md]: done

**Comment by domesticmouse** [comment]
And updated most of the templates to 2.4.

**Comment by domesticmouse** [comment]
PTAL @kwalrath

**Pull request merged**